mlflow/data/pandas_dataset.py

```diff
@@ -130,7 +145,9 @@ def to_pyfunc(self) -> PyFuncInputsOutputs:
         else:
             return PyFuncInputsOutputs(self._df)
```

Comment:
For L136-146: @prithvikannan The `to_pyfunc` method uses `targets` as an output. With the new introduced `predictions` column, shall we replace `targets` with `predictions` here? Besides, the docstring says this is used in `mlflow.evaluate()`. It seems that with the new `predictions` column. The definition of `PyFuncInputsOutputs(inputs, outputs)` should be updated to `PyFuncInputsOutputs(inputs, outputs, labels)` or something appropriate?

Reply:
@liangz1 this is very cool and makes perfect sense to add to MLflow datasets.